File: Data Structures & Algorithms/max-water-container/submission-1.go
func maxArea(heights []int) int {
	max := 0
	arr_len := len(heights)
	start, end := 0, arr_len-1
	for start < end {
		temp_max := min(heights[start], heights[end]) * (end - start)
		if temp_max > max {
			max = temp_max
		}
		if heights[start] < heights[end] {
			start++
		} else {
			end--
		}
	}
	return max
}

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-1.java
class Solution {
    public int largestRectangleArea(int[] heights) {
        int maxArea = 0, n = heights.length;
        for(int i = 0; i < n ; i++){
            int height = heights[i];
            int left = i, right = i;
            while(left - 1 >= 0 && heights[left - 1] >= heights[i]) {
                left--;
            }
            while(right + 1 < n && heights[right + 1] >= heights[i]) {
                right++;
            }
            int area = height * (right - left + 1);
            maxArea = Math.max(maxArea, area);
        }
        return maxArea;
    }
}